robot_modeling/DQ_SerialManipulatorMDH.m

```diff
@@ -0,0 +1,305 @@
+% modified Denavit-Hartenberg parameters (MDH)
+%
+% Usage: robot = DQ_SerialManipulatorMDH(A)
+% - 'A' is a 5 x n matrix containing the Denavit-Hartenberg parameters
+%   (n is the number of links)
+%    A = [theta1 ... thetan;
+%            d1  ...   dn;
+%            a1  ...   an;
+%         alpha1 ... alphan;
+%         type1  ... typen]
+% where type is the actuation type, either DQ_JointType.REVOLUTE
+% or DQ_JointType.PRISMATIC
+% - The only accepted convention in this subclass is the 'modified' DH
+% convention.
+%
+% If the joint is of type REVOLUTE, then the first row of A will
+% have the joint offsets. If the joint is of type PRISMATIC, then the
+% second row of A will have the joints offsets.
+%
+% DQ_SerialManipulatorMDH Methods (Concrete):
+%       get_dim_configuration_space - Return the dimension of the configuration space.
+%       fkm - Compute the forward kinematics while taking into account base and end-effector's rigid transformations.
+%       plot - Plots the serial manipulator.
+%       pose_jacobian - Compute the pose Jacobian while taking into account base's and end-effector's rigid transformations.
+%       pose_jacobian_derivative - Compute the time derivative of the pose Jacobian.
+%       raw_fkm - Compute the FKM without taking into account base's and end-effector's rigid transformations.
+%       raw_pose_jacobian - Compute the pose Jacobian without taking into account base's and end-effector's rigid transformations.
+%       set_effector - Set an arbitrary end-effector rigid transformation with respect to the last frame in the kinematic chain.
+% See also DQ_SerialManipulator.
+
+% (C) Copyright 2020-2022 DQ Robotics Developers
+%
+% This file is part of DQ Robotics.
+%
+%     DQ Robotics is free software: you can redistribute it and/or modify
+%     it under the terms of the GNU Lesser General Public License as published
+%     by the Free Software Foundation, either version 3 of the License, or
+%     (at your option) any later version.
+%
+%     DQ Robotics is distributed in the hope that it will be useful,
+%     but WITHOUT ANY WARRANTY; without even the implied warranty of
+%     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%     GNU Lesser General Public License for more details.
+%
+%     You should have received a copy of the GNU Lesser General Public License
+%     along with DQ Robotics.  If not, see <http://www.gnu.org/licenses/>.
+%
+% DQ Robotics website: dqrobotics.github.io
+%
+% Contributors to this file:
+%     1. Bruno Vihena Adorno (adorno@ieee.org)
+%        Responsible for the original implementation in file SerialManipulator.m 
+%        [bvadorno committed on Apr 10, 2019] (bc7a95f)
+%        (https://github.com/dqrobotics/matlab/blob/bc7a95f064b15046f43421d418946f60b1b33058/robot_modeling/DQ_SerialManipulator.m).
+%
+%     2. Murilo M. Marinho (murilo@nml.t.u-tokyo.ac.jp)
+%        - Created the file DQ_SerialManipulatorDH at #56
```

Comment:
- Reorganized the code by moving the implementation of SerialManipulator.m to the file DQ_SerialManipulatorDH at #56.

Reply:
[Implemented](https://github.com/dqrobotics/matlab/pull/75/commits/423e95fff8a38a0a0febf5a92a32a2c6c6fff8b6).